### PR TITLE
pin/biometrics news item

### LINKED ISSUE #4975: Reminder to configure pin biometrics
As a user i want to be reminded that I can secure my login credentials with pin/biometrics so that I am aware that this functionality exists.

## AC

- Only show the news item on mobile devices (even if decision was already made)
- Show the news item from the beginning to everyone (no delay)
- The following news item buttons are shown: "Secure Now", "Later", "No, thanks" (show error if they did not store credentials and select "Secure Now")
- "Secure Now" opens the credentials encryption dialog
- "Later" will remove the item from the news dialog but show it again
- "No, thanks" will remove the item and not show it again
- If the user selects credentials encryption show a dialog to rate the app
- It is possible for users to provide a rating text.
- The old 'Credentials encryption mode' is renamed to 'Unlock method'
- Rename the "Automatic" options to "None" or "Device lock"

### Proposed text:
Secure your app with pin or biometrics!

Tutanota focuses on security and privacy. So here is a little reminder that you can secure your login with PIN or biometrics like fingerprint or face ID. Just store the password in the app and then activate biometrics under Settings -> Login -> Unlock method.

Do you love the security and privacy you get with Tutanota? Then rate our apps now:

    iOS: https://apps.apple.com/app/tutanota/id922429609
    Android: https://play.google.com/store/apps/details?id=de.tutao.tutanota

By rating our app, you help us fight Big Tech dominance. Thank you very much!

### Biometrics? I barely know 'er
We need a less technical term. Maybe we can customize it depending on the device to show fingerprint/face id

## Test Notes:
* [ ] Create the pin/biometrics news item in the admin client (Usage -> News)
  * [ ] name: `pinBiometrics`
  * [ ] choose because of testing reasons a small repetition period
  * [ ] goes live at a specific time: no
  * [ ] rest: default
* [ ] Confirm that the news item will not be shown in any browser (mobile and desktop) and also not in the desktop client
* [ ] test the following on the iOS and android app
  * [ ] Log in to some account on a device that has not configured any unlock method yet
    * [ ] confirm that the news item is being displayed
    * [ ] confirm that on android the link to the playstore is displayed and on iOS the link to the appstore
    * [ ] click on later, open the news list again after the repetition period you earlier defined and make sure, that the news item is displayed again
    * [ ] click on "No thanks" and make sure that the news item will not appear again, even after the repetition period
    * [ ] create the news item again in the admin client and make sure that it appears on the device again after logging out and back in again
    * [ ] click "Secure now" and confirm that the unlock method dialog opens
    * [ ] select some unlock method and confirm that the news item will not appear again, even after the repetition period (also try to log out of the account, restart app, delete credentials and log in back again -> should still not be displayed again)
  * [ ] login to some other account and enable for example the password unlock method
    * [ ] log out and back in again (should prompt you to enter password)
    * [ ] the news item is displayed event though unlock method is already set

### PR DESCRIPTION
**Pull request by nokhub**
the device config offers now a possibility to store which news item were already acknowledged on this specific device. News items can use this information to decide whether it is displayed or not.

close #4975

Co-authored-by: nig <nig@tutao.de>

### CODE REVIEW

File: test/tests/misc/NewsModelTest.ts [hunk 1]
```diff
@@ -24,8 +26,9 @@ o.spec("NewsModel", function () {
 
 	o.beforeEach(function () {
 		serviceExecutor = object()
+		deviceConfig = object()
 
-		newsModel = new NewsModel(serviceExecutor, async () => new DummyNews())
+		newsModel = new NewsModel(serviceExecutor, deviceConfig, async () => new DummyNews())
```
Comment (charlag) [inline test/tests/misc/NewsModelTest.ts:31]: real `deviceConfig`? In this economy?

File: src/misc/TranslationKey.ts [hunk 2]
```diff
@@ -1017,6 +1020,10 @@ export type TranslationKeyType =
 	| "phishingReport_msg"
 	| "phone_label"
 	| "photoLibraryUsageDescription_msg"
+	| "pinBiometrics_action"
+	| "pinBiometrics_msg1"
```
[the remark this answers is not in the document]
Comment (nokhub) [reply src/misc/TranslationKey.ts]: that's true, I didn't think about it

File: src/misc/news/NewsModel.ts [hunk 3]
```diff
@@ -11,7 +12,11 @@ export class NewsModel {
 	liveNewsIds: NewsId[] = []
 	liveNewsListItems: Record<string, NewsListItem> = {}
 
-	constructor(private readonly serviceExecutor: IServiceExecutor, private readonly newsListItemFactory: (name: string) => Promise<NewsListItem | null>) {}
+	constructor(
+		private readonly serviceExecutor: IServiceExecutor,
+		private readonly deviceConfig: DeviceConfig,
```
[the remark this answers is not in the document]
Comment (nokhub) [reply src/misc/news/NewsModel.ts]: yeah that's better

**Review by charlag — APPROVED** [review]
I think I misunderstood that you've been mocking it before too (even though I looked together with @ganthern), sorry, but good to have a little interface anyway!

**Pull request merged**